testnet/stacks-node/src/nakamoto_node/stackerdb_listener.rs

```diff
@@ -487,30 +480,32 @@ impl StackerDBListenerComms {
 
     /// Get the status for `block` from the Stacker DB listener.
     /// If the block is not found in the map, return an error.
-    /// If the block is found, call `condition` to check if the block status
-    /// satisfies the condition.
-    /// If the condition is satisfied, return the block status as
-    ///   `Ok(Some(status))`.
-    /// If the condition is not satisfied, wait for it to be satisfied.
+    /// If the block is found, return it.
     /// If the timeout is reached, return `Ok(None)`.
-    pub fn wait_for_block_status<F>(
+    pub fn wait_for_block_status(
         &self,
         block_signer_sighash: &Sha512Trunc256Sum,
+        block_status_tracker: &mut BlockStatus,
+        rejections_timer: std::time::Instant,
+        rejections_timeout: Duration,
         timeout: Duration,
-        condition: F,
```

Comment:
Why not leave the `StackerDBListener` and `StackerDBListenerComms` as-is and handle the timeout in `SignerCoordinator::get_block_status`?

Reply:
Definitely no reason, i changed it for a previous implementation that i trashed away. I am going to restore it.